pkg/container/cri/yaml.go

```diff
@@ -28,17 +28,24 @@ Custom YAML (de)serialization for these types
 // UnmarshalYAML implements custom decoding YAML
 // https://godoc.org/gopkg.in/yaml.v3
 func (m *Mount) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	type Alias Mount
-	aux := &struct {
-		Propagation string `yaml:"propagation"`
-		*Alias
-	}{
-		Alias: (*Alias)(m),
+	// this is basically Mount, except Propogation is a string for further parsing
+	type MountYaml struct {
```

Comment:
doing this gives pretty good errors though (type names show up in errors regarding keys)